packages/boot/test/bootstrapTests/benchmarkerator.md

```diff
@@ -0,0 +1,171 @@
+which, by convention, goes in the `benchmarks` directory of your package,
+parallel to the conventional `test` directory (as of this writing very few
+packages have such a `benchmarks` directory, so you might need to add it
+yourself).[^2]
+
+[^2]: In the fullness of time, the `yarn bench` command will be updated to run
+    all the benchmarks found in the `benchmarks` directory, much as it now runs
+    the older, Ava-based benchmarks that it looks for in the `test` directory.
+    Those will themselves be deprecated and/or moved into `benchmarks`.
+    However, as of this writing that has not yet been done.
+
+The first thing a benchmark should do is import The Benchmarkerator:[^3]
+```
+import { bench } from '@agoric/benchmark';
+```
+
+[^3]: As of this writing The Benchmarkerator actually lives in
+    `@agoric/boot/test/boostrapTests/benchmarkerator.js`.  This should be
+    changed soon, hopefully before the first PR containing this writeup is
+    landed, in which case you will never see this footnote.
+
+Note that this importation usually should be the very first thing you do, much
+as you typically import `@agoric/swingset-vat/tools/prepare-test-env-ava.js` or
+`@agoric/zoe/tools/prepare-test-env-ava.js` or the like as the first thing in a
+test implementation.  The exception, as with tests, is imports that bring in any
+kind of pre-lockdown functionality that you will be needing to use; however,
+that should be necessary only in very specialized circumstances.
+
+In a manner similar to the way tests are packaged for Ava, a benchmark file can
+actually contain multiple benchmark tests.  Each should be declared with a
+statement of the form:
+
+`bench.addBenchmark(label: string, benchmark: Benchmark);`
+
+where `label` is a string labelling the benchmark (much as individual tests are
+labelled in Ava), and `benchmark` is an object describing the benchmark itself
+(which will be described in detail below)
+
+After defining one or more benchmarks with `addBenchmark` you must then end by
+invoking:
+
+```await bench.run(name);```
+
+where `name` is a string naming the benchmark run for purposes of error and
+result reporting.
+
+#### The `Benchmark` object
+
+The `benchmark` object may have the following properties, all of which are
+optional except for `executeRound`:
+
+`setup?: (context: BenchmarkContext) => Promise<Record<string, unknown> | undefined>`
+
+  An optional async method to perform any pre-run setup that you need to do. It
+  may optionally return a configuration record containing any benchmark-specific
+  information you desire.  This will be provided as part of the context object
+  in subsequent calls into the benchmark.
+
+`executeRound: (context: BenchmarkContext, round: number) => Promise<void>`
+
+  A required async method which is called to actually execute one round of the
+  benchmark.  The `round` argument is the number of the benchmark round
+  that this call to `executeRound` is being asked to execute.
```

Comment:
I was about to write "it's 1-based, of course; 0-based wouldn't make sense here", but then I realized I wasn't actually 100% sure myself and had to look at the code (for the record, it's 1-based, because it's a count).  So I'll update the doc to make this clear.